packages/cli/src/e2e.test.ts

```diff
@@ -0,0 +1,179 @@
+import { cli } from './cli';
+import { randomUUID } from 'crypto';
+
+describe('e2e', () => {
+    const out = 'generated/synthql/e2e/tests';
+    const configFile = 'src/tests/synthql.config.json';
+
+    beforeAll(() => {
+        fs.rmSync(out, {
+            force: true,
+            recursive: true,
+        });
+    });
+
+    beforeEach(() => {
+        vitest.resetModules();
+        vitest.resetAllMocks();
+    });
+
+    test('Write schema to files', async () => {
+        const prettierOptions = await prettier.resolveConfig(
+            '../../.prettier.config.js',
+        );
+
+        expect(prettierOptions).not.toBe(null);
+
+        // Generate in `cli` package
+        const cliOutDir = '../cli/src/tests';
+
+        if (!fs.existsSync(cliOutDir)) {
+            fs.mkdirSync(cliOutDir, { recursive: true });
+        }
+
+        fs.writeFileSync(
+            path.join(cliOutDir, 'synthql.dev.config.json'),
+            await prettier.format(JSON.stringify(configFileSchema, null, 2), {
+                parser: 'json',
+                ...prettierOptions,
+            }),
+        );
+
+        // Generate in `docs` package
+        const docsOutDir = '../docs/static/schemas';
+
+        if (!fs.existsSync(docsOutDir)) {
+            fs.mkdirSync(docsOutDir, { recursive: true });
+        }
+
+        fs.writeFileSync(
+            path.join(docsOutDir, 'synthql.config.json'),
+            await prettier.format(JSON.stringify(configFileSchema, null, 2), {
+                parser: 'json',
+                ...prettierOptions,
+            }),
+        );
+    }, 100_000);
+
+    test(
+        `Generate schema types with a supplied config file option, --configFile`,
+        async () => {
+            const outDir = `${out}/${Date.now()}-${randomUUID()}`;
+
+            const argv = [
+                'node',
+                'generate-with-default-schema.js',
+                'generate',
+                `--configFile=${configFile}`,
+                `--out=${outDir}`,
+            ];
+
+            // Compose file path
+            const outFilePath = path.join(outDir, `db.ts`);
+
+            // Check that file with the name doesn't exist yet
+            expect(!fs.existsSync(outFilePath)).toEqual(true);
+
+            // Generate schema file
+            const output = await cli({ argv, exit: () => {} });
+
+            // Check that the file now exists
+            expect(fs.existsSync(outFilePath)).toEqual(true);
+
+            // Verify that the option data is correct
+            expect(output.configFile).toEqual(configFile);
+
+            // Verify that the passed attribute overrides were applied
+            const schemaFilePath = path.join(outDir, `schema.ts`);
+
+            const file = await import(schemaFilePath);
+            const schema = file.schema.properties;
+
+            const schemaDefOverrides = output.schemaDefOverrides as any;
+
+            for (const qualifiedTableName in schemaDefOverrides) {
```

Comment:
wow what's happening here? what are you trying to assert?

Reply:
Usually it's not a good idea to have too much logic inside your tests, as bugs can creep into the logic over time.

Can we simplify this logic? Why can't we just do a single big `expect`?

Reply:
I will look into simplifying it, but a big expect might not be possible because the objects don't have the same shape